### DISCUSSION TITLE
Module not found: Can't resolve '@web3modal/wagmi/react

### DISCUSSION THREAD

**Discussion by amit-atr958** [WalletConnect/walletconnect-monorepo#3889] [Developer Support] [answered]
i am getting this error
import { useWeb3Modal } from "@web3modal/wagmi/react";

export default function ConnectButton() {
  // 4. Use modal hook
  const { open } = useWeb3Modal();

  return (
    <>
      <button onClick={() => open()}>Open Connect Modal</button>
      <button onClick={() => open({ view: "Networks" })}>
        Open Network Modal
      </button>
    </>
  );
}

**Comment by boidushya** [comment]
Have you installed all relevant dependencies as mentioned in docs?

**Reply by damn-ice** [reply]
@boidushya I am also experiencing same issue when I try to start my react app, I Installed all dependencies as stated on the docs.
![walletconnect-issue](https://github.com/WalletConnect/walletconnect-monorepo/assets/65536487/0fb5d27f-745e-4b26-a91e-0751ae7768e8)

**Reply by amit-atr958** [reply]
i have installed all relevant dependencies

**Comment by damn-ice** [comment]
@amit-atr958 Were you able to resolve the issue?

**Reply by amit-atr958** [reply]
yes

**Comment by glitch-txs** [comment]
Could you share a public repository we can reproduce this issue?

**Reply by glitch-txs** [reply]
@amit-atr958

**Reply by taildfoobla** [reply]
@glitch-txs 
I have the same issue and this is my repository.
https://github.com/taildfoobla/walletconnect-modal-wagimi-v1
Thank you for helping.

**Comment by damn-ice** [comment]
@glitch-txs Sorry for the delayed response, I had to complete some other task.

I tried starting all over with steps as stated in the documentations

`yarn add @web3modal/wagmi wagmi viem`

seems our project has some issues with dependecies, I don't know if that could be the reason(screenshots below) but I update typescript to 5.0.4 but still I got the issue.
![wallet_connect_dependency_issues](https://github.com/WalletConnect/walletconnect-monorepo/assets/65536487/5999f22b-858e-47b0-9c20-ec4223385855)

Also, I can see the `wagmi/react` in `node_module` folder (screenshot below).

![wagmi-react](https://github.com/WalletConnect/walletconnect-monorepo/assets/65536487/50f690f5-df47-4445-9454-1151f05a2f08)

Thanks.

**Comment by Evirtual** [comment]
I am also experiencing the same issue, but in jest tests, when mocking "@web3modal/wagmi/react". The same goes for "@web3modal/wagmi/react/config".

**Comment by amit-atr958** [comment, marked as answer]
@damn-ice update your  react app version latest and node version is 18+ version

**Reply by RPriest** [reply]
What version of node work perfectly, I'm having same issue too

**Reply by fukemy** [reply]
i am use node 20

**Comment by prabhatkashyap902** [comment]
hey, any solution?? experiencing the same issue,

**Comment by sudharsansubash2002** [comment]
facing same issue. Is there any solutiong for this?

**Reply by amit-atr958** [reply]
@sudharsansubash2002  Update to the latest version of your React app and node version is 18+.

**Comment by fukemy** [comment]
any solution? I got this error with nextjs:
<img width="1018" alt="Ảnh màn hình 2024-08-21 lúc 11 21 53" src="https://github.com/user-attachments/assets/57a88481-8fd5-47bd-9bd6-be24a0c0e810">
